### PR TITLE
Add a regression test [RIPD-1040]

### PR DESCRIPTION
**Pull request by scottschurr**
As reported in RIPD-1040, part way into the development of ed25519 key support a secp256r1 public key could cause rippled to seg fault.  This test reproduces the seg fault if the fix is not present and passes when the fix is in place.

reviewers: @ximinez, @nbougalis

### CODE REVIEW

**Comment by nbougalis** [comment]
Test looks good so :+1: on that from me. Consider rewording the commit message to:

    Test handling of secp256r1 signatures (RIPD-1040)

**Comment by scottschurr** [comment]
Removed unnecessary comment and corrected commit message.

File: src/ripple/app/tests/Regression_test.cpp [hunk 1]
```diff
@@ -111,10 +112,57 @@ struct Regression_test : public beast::unit_test::suite
...
+    void testSecp256r1key ()
+    {
+        testcase("Signing with a secp256r1 key should fail gracefully");
+        using namespace jtx;
+        Env env(*this);
+
+        // Test case we'll use.
+        auto test256r1key = [&env] (Account const& acct)
+        {
+            auto const baseFee = env.app().config().FEE_DEFAULT;
```
Comment (ximinez) [inline src/ripple/app/tests/Regression_test.cpp]: `env.current()->fees().base`

Comment (scottschurr) [inline src/ripple/app/tests/Regression_test.cpp]: Good catch.  Thanks.

Comment (ximinez) [inline src/ripple/app/tests/Regression_test.cpp]: Yer welcome.

**Comment by ximinez** [comment]
Nice test. Verified that rippled to crashes in this test if the mentioned commit is reverted.
:+1:

**Pull request closed without merging**